src/app/inventory/store/selectors.ts

```diff
@@ -0,0 +1,121 @@
+      bucketsWithClassifieds.add(i.bucket.hash);
+    }
+  }
+  return bucketsWithClassifieds;
+}
+
+export interface StorePowerLevel {
+  /** average of your highest gear, even if not equippable at the same time */
+  maxGearPower: number;
+  /** currently represents the power level bonus provided by the Seasonal Artifact */
+  powerModifier: number;
+  /** maxGearPower + powerModifier. the highest PL you can get your inventory screen to show */
+  maxTotalPower: number;
+
+  /** the highest-power items per bucket, even if not equippable at the same time */
+  highestPowerItems: DimItem[];
+
+  /** maxGearPower and maxTotalPower can come with various caveats */
+  problems: {
+    /** this stat may be inaccurate because it relies on classified items */
+    hasClassified: boolean;
+    /** mutually excluded exotics are included in the max possible power */
+    notEquippable: boolean;
+    /** this character is in danger of dropping at a worse Power Level! another character is holding their best item(s) */
+    notOnStore: boolean;
+  };
+}
+
+const allPowerLevelsSelector = createSelector(
+  storesSelector,
+  allItemsSelector,
+  (stores, allItems) => {
+    const bucketsWithClassifieds = getBucketsWithClassifiedItems(allItems);
+
+    const levels: {
+      [storeId: string]: StorePowerLevel;
+    } = {};
+
+    for (const store of stores) {
+      if (store.isVault) {
+        continue;
+      }
+
+      const { equippable, unrestricted } = maxLightItemSet(allItems, store);
+
+      // ALL WEAPONS count toward your drops. armor on another character doesn't count.
+      // (maybe just because it's on a different class? who knows. can't test.)
+      const dropPowerItemSet = maxLightItemSet(
+        allItems.filter((i) => i.bucket.inWeapons || i.owner === 'vault' || i.owner === store.id),
+        store
+      ).unrestricted;
+      const dropPowerLevel = getLight(store, dropPowerItemSet);
+
+      const unrestrictedMaxGearPower = getLight(store, unrestricted);
+      const equippableMaxGearPower = getLight(store, equippable);
+
+      const notEquippable = unrestrictedMaxGearPower !== equippableMaxGearPower;
+      const notOnStore = dropPowerLevel !== unrestrictedMaxGearPower;
+      const hasClassified = hasAffectingClassified(unrestricted, bucketsWithClassifieds);
+      const artifactPower = getArtifactBonus(store);
+
+      levels[store.id] = {
+        maxGearPower: unrestrictedMaxGearPower,
+        powerModifier: artifactPower,
+        maxTotalPower: unrestrictedMaxGearPower + artifactPower,
+        highestPowerItems: unrestricted,
+        problems: {
+          hasClassified,
+          notEquippable,
+          notOnStore,
+        },
+      };
+    }
+
+    return levels;
+  }
+);
+
+export const powerLevelSelector = createSelector(
+  allPowerLevelsSelector,
+  (_state: RootState, storeId?: string) => storeId,
+  (allPowerLevels, storeId) => (storeId !== undefined ? allPowerLevels?.[storeId] : undefined)
+);
```

Comment:
I might not use `createSelector` here, since just accessing a property of `allPowerLevels` is not expensive.